clio-client/src/main/scala/org/broadinstitute/clio/client/webclient/ClioWebClient.scala

```diff
@@ -59,101 +56,51 @@ class ClioWebClient(
     dispatchRequest(HttpRequest(uri = "/health"))
   }
 
-  def getSchemaWgsUbam(
-    implicit bearerToken: OAuth2BearerToken
-  ): Future[HttpResponse] = {
+  def getSchema(
+    transferIndex: TransferIndex
+  )(implicit credentials: HttpCredentials): Future[HttpResponse] = {
     dispatchRequest(
-      HttpRequest(uri = "/api/v1/wgsubam/schema")
-        .addHeader(Authorization(credentials = bearerToken))
+      HttpRequest(uri = s"/api/v1/${transferIndex.urlSegment}/schema")
+        .addHeader(Authorization(credentials = credentials))
     )
   }
 
-  def addWgsUbam(
-    input: TransferWgsUbamV1Key,
-    transferWgsUbamV1Metadata: TransferWgsUbamV1Metadata
-  )(implicit bearerToken: OAuth2BearerToken): Future[HttpResponse] = {
+  def upsert[T](transferIndex: TransferIndex, key: TransferKey, metadata: T)(
```

Comment:
Instead of `T`, can we use `transferIndex.metadataType`?

Reply:
Actually, I think this won't work in current Scala unless we break the params into two lists because of how scoping for arguments works:
```scala
def upsert(transferIndex: TransferIndex)(key: transferIndex.KeyType, metadata: transferIndex.MetadataType)
```
Up to you if you want to try locking it down.